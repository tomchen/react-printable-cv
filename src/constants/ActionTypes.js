export const CHANGE_LANG = 'CHANGE_LANG'

export const RENDER_DATA = 'RENDER_DATA'

export const IMPORT_JSON = 'IMPORT_JSON'
