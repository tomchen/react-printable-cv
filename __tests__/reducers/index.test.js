import reducer from 'Src/reducers'
import * as types from 'Src/constants/ActionTypes'

const initialState = {
  lang: null,
  userData: {},
  projectData: {},
}

describe('Initial state', () => {
  it('is correct', () => {
    const s = expect(reducer(undefined, {}))
    s.toEqual(initialState)
    s.toMatchSnapshot()
  })
})

describe('reducer', () => {
  it('should handle CHANGE_LANG', () => {
    const lang = 'zh-cn'

    const s = expect(
      reducer(initialState, {
        type: types.CHANGE_LANG,
        lang,
      }),
    )
    s.toEqual({
      lang,
      userData: {},
      projectData: {},
    })
    s.toMatchSnapshot()
  })

  it('should handle RENDER_DATA', () => {
    const data = { userData: 6, projectData: 'f' }

    const s = expect(
      reducer(initialState, {
        type: types.RENDER_DATA,
        data,
      }),
    )
    s.toEqual({
      lang: null,
      ...data,
    })
    s.toMatchSnapshot()
  })

  it('should handle IMPORT_JSON', () => {
    const data = { userData: 'n', projectData: 7 }

    const s = expect(
      reducer(initialState, {
        type: types.IMPORT_JSON,
        data,
      }),
    )
    s.toEqual({
      lang: null,
      ...data,
    })
    s.toMatchSnapshot()
  })
})
