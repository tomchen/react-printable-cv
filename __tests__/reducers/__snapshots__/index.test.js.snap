// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial state is correct 1`] = `
Object {
  "lang": null,
  "projectData": Object {},
  "userData": Object {},
}
`;

exports[`reducer should handle CHANGE_LANG 1`] = `
Object {
  "lang": "zh-cn",
  "projectData": Object {},
  "userData": Object {},
}
`;

exports[`reducer should handle IMPORT_JSON 1`] = `
Object {
  "lang": null,
  "projectData": 7,
  "userData": "n",
}
`;

exports[`reducer should handle RENDER_DATA 1`] = `
Object {
  "lang": null,
  "projectData": "f",
  "userData": 6,
}
`;
